src/video/ogc/SDL_ogcvideo.c

```diff
@@ -145,6 +153,32 @@ static void add_supported_modes(SDL_VideoDisplay *display, u32 tv_format)
     init_display_mode(&mode, &s_mode320);
     SDL_AddDisplayMode(display, &mode);
 
+    // libogc uses 640 for the viWidth, but this does not work well for Widescreen
+    // as such we extend the viWidth to 704 on a new videomode
+    // TODO: Currently ony added on wii if it's widescreen,
+    // but should work on GC and a 4:3 Wii
+    // testing needed
+
+#ifdef __wii__
+    if (CONF_GetAspectRatio() == CONF_ASPECT_16_9) {
+        memcpy(&s_mode704, gx_modes[1], sizeof(s_mode704));
+        s_mode704.viWidth = 704;
+
+        // set Center point
+        if (&s_mode704 == &TVPal576IntDfScale || &s_mode704 == &TVPal576ProgScale) {
```

Comment:
This should be safer, in case we add more modes later:

```suggestion
        if (VI_FORMAT_FROM_MODE(s_mode704.viTVMode) == VI_PAL) {
```